content/3_Silver/Prefix_Sums.mdx

```diff
@@ -307,3 +307,122 @@ for i in range(Q):
+  <Quiz.Question>
+    You are given the following array: $A = [2, 5, 7, 1, 4]$. What is the prefix sum array of $A$?
+    <Quiz.Answer>
+      $P = [2, 3, 2, -6, 3]$
+      <Quiz.Explanation>
+        Incorrect. This is known as the difference array, where the value of each index is determined by subtracting the two elements in $A$. This is the inverse operation of the prefix sum array, meaning that taking the prefix sum of the difference array or vice versa will result in the original array $A$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      $P = [2, 7, 14, 15, 19]$
+      <Quiz.Explanation>
+        Correct! The value at each index is equal to the sum of the values in $A$ before and at the current index.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $P = [0, 7, 12, 8, 5]$
+      <Quiz.Explanation>
+        Incorrect. Recall that the prefix sum array is the sum of **all** previous element, not just the element directly before it. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $P = [19, 17, 12, 5, 4]$
+      <Quiz.Explanation>
+        Incorrect. This is the "suffix sum" array, which is the prefix sum array computed in reverse. In general, it serves the same purpose as its counterpart so we rarely use it. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    We are given the prefix sum array $P = [2, 7, 14, 15, 19]$. Find the sum of values _of the original array_ for the ranges of indexes $[1, 4]$ and $[2, 5]$ (inclusive, 1-indexed).
+    <Quiz.Answer>
+      13; 12
+      <Quiz.Explanation>
+        Incorrect. Recall that to get the sum of some range of values $[l, r]$, we need to calculate $P[r] - P[l - 1]$ instead of $P[r] - P[l]$. In the second formula, we do not include the value at index $l$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      17; 26
+      <Quiz.Explanation>
+        Incorrect. To get the sum of some range of values $[l, r]$, we need to calculate $P[r] - P[l - 1]$ instead of $P[r] + P[l]$. .
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      15; 17
+      <Quiz.Explanation>
+        Correct! The sum of the values of some subarray $[l, r]$ is determined by the formula $P[r] - P[l - 1]$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      15; 21
+      <Quiz.Explanation>
+        Incorrect. Recall that to get the sum of some range of values $[l, r]$, we need to calculate $P[r] - P[l - 1]$ instead of $P[r] + P[l - 1]$. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    **Challenge Problem**: With the idea of prefix sums in mind, at first glance it may seem logical to apply this method to products. Instead of adding, we multiply, and instead of subtracting, we divide. Assume that we calculate the answer modulo a large prime ($10^9 + 7$, for instance). What additional challenges may we face?
```

Comment:
```suggestion
    **Challenge Problem**: With the idea of prefix sums in mind, at first glance it may seem logical to apply this method to return range products in constant time. Instead of adding, we multiply, and instead of subtracting, we divide. Assume that we calculate the answer modulo a large prime ($10^9 + 7$, for instance). What additional challenges may we face?
```